File: be/ClinicManager/src/main/java/com/clinic/clinicmanager/model/Sale.java
package com.clinic.clinicmanager.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "sale")
public class Sale {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "sale_id")
    private List<SaleItem> items = new ArrayList<>();

    @Column(nullable = false)
    @Builder.Default
    private Double totalNet = 0.0;
    @Column(nullable = false)
    @Builder.Default
    private Double totalVat = 0.0;
    @Column(nullable = false)
    @Builder.Default
    private Double totalValue = 0.0;
}

File: fe/ClinicManager/src/models/sale.ts
import { Product } from "./product";
import { NewVoucher, Voucher } from "./voucher";


export interface Sale {
    id:number;
    items: SaleItem[];
    totalNet: number;
    totalVat: number;
    totalValue: number;
}

export interface NewSale {
    items: NewSaleItem[];
}

export interface SaleItem {
    id: number;
    product: Product | null;
    voucher: Voucher | null;
    name: string;
    netValue: number;
    vatValue: number;
    price: number;
}

export interface NewSaleItem {
    product?: Product;
    voucher?: NewVoucher;
    name?: string;
    price: number;
}